modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -837,10 +925,27 @@ struct InferList2: public cv::detail::KernelTag {
         // FIXME: this is filtering not done, actually! GArrayDesc has
         // no hint for type!
         const auto &mm_0   = in_metas[0u];
-        const auto &meta_0 = util::get<cv::GMatDesc>(mm_0);
-        GAPI_Assert(   !meta_0.isND()
-                    && !meta_0.planar
-                    && "Only images are supported as the 0th argument");
+        switch (in_metas[0u].index()) {
+            case cv::GMetaArg::index_of<cv::GMatDesc>(): {
+                const auto &meta_0 = util::get<cv::GMatDesc>(mm_0);
+                GAPI_Assert(   !meta_0.isND()
+                        && !meta_0.planar
+                        && "Only images are supported as the 0th argument");
+                break;
+            }
+            case cv::GMetaArg::index_of<cv::GFrameDesc>(): {
+                // FIXME: Is there any validation for GFrame ?
```

Comment:
Maybe you need to check color formats here? If there are similar `case`s with no checks for `GFrameDesc` lets just left this as is

Reply:
Ok.

Reply:
Is it done?

Reply:
Added check for GFrameDesc parameters. I think it should be a function (analogue of isND()) because IE backend should do similar checks. Create a ticket for this?